device.go

```diff
@@ -274,7 +274,7 @@ func createDevice(devInfo DeviceInfo) Device {
 
 	if isVFIO(path) {
 		return newVFIODevice(devInfo)
-	} else if isBlock(path) {
+	} else if isBlock(path) || devInfo.DevType == "b" {
```

Comment:
This should be part of `isBlock()`. Providing `devInfo` through `isBlock` should be enough.